gencert-private.sh

```diff
@@ -1,15 +1,35 @@
 #!/bin/sh
 
+csr="csr.csr"
+key="key.pem"
+
+# check organizationIdentifier valid value (VATID-XXX... or CF:IT-XXX..)
+
+if echo "$ORGANIZATION" | grep -qe 'CF:IT-*' || echo "$ORGANIZATION" | grep -qe 'VATID-*'; then  
```

Comment:
`VATID-*` is partially correct and too permissive. According to the specs, the format is

`VAT` + `country code in alpha2` + `-` + `partita iva (11 digits)`

Therefore, I would "grep" with the following regex

`^VAT[A-Z]{2}-\d{11}$`

About `CF:IT-*`, I think it also is too permissive. I would use something like that

`^CF:IT-[\d\w]{16}$`

So finally, the check could result in something like that

```
if [ $(echo ${ORGANIZATION} | grep -cP "^(CF:IT-[\d\w]{16}|VATIT-\d{11})$") -ne 1 ]; then
   echo "Invalid format for ORGANIZATION"
   exit 1
fi

Reply:
In order to be aligned with the `gencert-public.sh` script, the `ORGANIZATION` variable should be renamed in `ORGANIZATION_IDENTIFIER`.